### ISSUE TITLE
[Suggestion] First check all pkgbuilds and start building packages, then install

### ISSUE THREAD

**Issue by JuniorJPDJ** [trizen/trizen#89]
Hi,
I think that changing order of working would make trizen more convenient.
You can check if everything is ok with pkgbuilds, let everything build in background and when you think it's finished look at this terminal window and accept instalation of all built packages.
PS. I love your feature to display PKGBUILDs before entering edit mode.

**Comment by flying-sheep** [comment]
That would even allow a `make`-like `-j` option for parallel building

**Comment by Noxygen13** [comment]
I'm here to show support of this idea. That's what I really liked about pacaur and trizen's looking great except for this feature. I'd like to be able to put it in background and don't have to input anything until my updates are done. Also to go with the idea of @flying-sheep, it'd be awesome for trizen to be able to download a certain number of packages in parallel as many gits and repos can't saturate high speed connexions. This would make the update process way faster. I'm pretty sure installing in parallel is a terrible idea though... but it could start installing as soon as a package is downloaded and continue the other downloads in the background

**Comment by mbloms** [comment]
@Noxygen13 actually you can't install multiple packages at once. `pacman` won't allow it.

This would be my dream:

1. Fetch repos in parallel
1. Start install all makedepends
1. Review/Accept PKGBUILDS as depends are installing
2. Build in parallel after all depends are installed
2. Install all with `pacman -U` when all packages are built

**Comment by Noxygen13** [comment]
That'd be awesome @mbloms indeed, except building in parallel which can become really messy but alas it's only a dream.

A good middleground would be to download all the PKGBUILDs at the same time, ask all the questions then and after that not require any user input (like pacaur for instance), at this point I don't think a lot of people would care if it's not parallelizing downloads or installs as it's already done to them, just a background task.

**Comment by mbloms** [comment]
@Noxygen13 why do you think parallelising builds wouldn't work? As long as you resolve all dependencies, it should be fine, wouldn't it? I often use makepkg manually and when I do, I regularly build multiple packages at once.

**Comment by Noxygen13** [comment]
@mbloms there can be issues with the RAM, hitting the swap is way more damaging than waiting for another compilation to finish. And it's nearly impossible to predict what amount of RAM one build process will take

**Comment by mbloms** [comment]
Good point.

**Comment by MarcJose** [comment]
I agree, atleast installing all packages with one command would be great especially in terms of long running hooks which right now run after each and every package.
